ui/app/AppLayouts/Chat/controls/community/HoldingsDropdown.qml

```diff
@@ -0,0 +1,277 @@
+import QtGraphicalEffects 1.13
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Controls.Validators 0.1
+import StatusQ.Core.Utils 0.1 as SQ
+
+import utils 1.0
+
+StatusDropdown {
+    id: root
+
+    property real tokenAmountValue: 0
+    property string tokenName: d.defaultTokenNameText
+    property url tokenImage: ""
+    property int operator: SQ.Utils.Operators.None
+    property bool withOperatorSelector: true
+
+    signal addToken(string tokenText, url tokenImage, int operator)
+
+    function reset() {
+        root.tokenAmountValue = 0
+        root.tokenName = d.defaultTokenNameText
+        root.tokenImage = ""
+        root.operator = SQ.Utils.Operators.None
+    }
+
+    QtObject {
+        id: d
+
+        readonly property bool ready: root.tokenAmountValue > 0 && root.tokenName !== d.defaultTokenNameText
+
+        // By design values:
+        readonly property int initialHeight: 232
+        readonly property int mainHeight: 256
+        readonly property int operatorsHeight: 96
+        readonly property int extendedHeight: 417
+        readonly property int defaultWidth: 289
+        readonly property int operatorsWidth: 159
+
+        property string defaultTokenNameText: qsTr("Choose token")
+
+        function selectInitState() {
+            if(root.withOperatorSelector)
+                loader.sourceComponent = operatorsSelectorView
+            else
+                loader.sourceComponent = tabsView
+        }
+    }
+
+    width: d.defaultWidth
+    height: d.initialHeight
+
+    contentItem: Loader {
+        id: loader
+        anchors.fill: parent
+        sourceComponent: root.withOperatorSelector ? operatorsSelectorView : tabsView
+
+        onSourceComponentChanged: {
+            if(sourceComponent == tokensExtendedView) {
+                root.height = Math.min(item.contentHeight + item.anchors.topMargin + item.anchors.bottomMargin, d.extendedHeight)
+                root.width = d.defaultWidth
+            }
+            else if(sourceComponent == operatorsSelectorView) {
+                root.height = d.operatorsHeight
+                root.width = d.operatorsWidth
+            }
+            else if(sourceComponent == tabsView && root.withOperatorSelector)  {
+                root.height = d.mainHeight
+                root.width = d.defaultWidth
+            }
+            else if(sourceComponent == tabsView && !root.withOperatorSelector)  {
+                root.height = d.initialHeight
+                root.width = d.defaultWidth
+            }
+        }
+    }
+
+    onOpened: d.selectInitState()
+    onClosed: root.reset()
+    onWithOperatorSelectorChanged: d.selectInitState()
+
+    Component {
+        id: tabsView
+
+        ColumnLayout {
+            anchors.fill: parent
+            anchors.margins: 8
+            anchors.topMargin: 16
+            spacing: 8
+            StatusIconTextButton {
```

Comment:
`StatusBackButton`

Reply:
Same comment than before, let me keep it as it is as our `StatusBackButton` looks different from the designs of this dropdown.

Reply:
Yup, good